File: 프로그래머스/unrated/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.Arrays;
import java.util.PriorityQueue;

class Solution {
    public int[] solution(int k, int[] score) {
        int[] answer = new int[score.length];
        PriorityQueue<Integer> queue = new PriorityQueue<>();
        for (int i = 0; i < score.length; i++) {
            queue.add(score[i]);
            if (queue.size() > k) {
                queue.poll();
            }
            answer[i] = queue.peek();
        }
        return answer;
    }
    /*
    * 우선순위 큐, "PriorityQueue"는 기본적으로 작은 값이 높은 우선순위를 가지도록 구현되어있다.
    * poll() 메서드를 호출하면 가장 작은 값이 제거되고.
    * peek() 호출하면 가장 작은 값이 반환된다.
    * 반대로 큰 값이 우선순위를 갖도록 하려면 PriorityQueue<Integer> queue = new PriorityQueue<>(Comparator.reverseOrder());
    */
        public static void main(String[] args) {
        Solution s = new Solution();
        int k = 3;
        int[] score = {10, 100, 20, 150, 1, 100, 200};
        int[] answer = s.solution(k,score);
        System.out.println("answer = " + Arrays.toString(answer));
    }
}

File: JavaScript/프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    let answer = [];
    let crwon = [];
    score.forEach(v => {
        if ( crwon.length < k) {
            crwon.push(v);
        } else if (v > crwon[0]) {
            crwon.shift();
            crwon.push(v);
        }
        crwon.sort((a, b) => a - b);
        answer.push(crwon[0]);
    });
    console.log();
    return answer;
}